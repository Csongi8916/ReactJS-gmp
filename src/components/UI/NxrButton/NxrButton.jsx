import React from 'react';
import './NxrButton.scss';

export default function NxrButton(props) {
  return (
    <React.Fragment>
      <button className="NrxBtn" onClick={props.click}>{props.children}</button>
    </React.Fragment>
  )
}
