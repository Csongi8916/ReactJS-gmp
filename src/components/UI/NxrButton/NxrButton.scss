$nrRedColor: #f65261;
$nrBlackColor: #232323;
$nrWhiteColor: #ffffff;

.NrxBtn {
  margin-left: 10px;
  background-color: $nrRedColor;
  color: #fff;
  cursor: pointer;
  width: 150px;
  border: 0;
  border-radius: 3%;
  outline: none;
  font-family: Montserrat, Helvetica, sans-serif;
  font-size: 18px;
  text-transform: uppercase;
  padding: 13px 10px;
}
