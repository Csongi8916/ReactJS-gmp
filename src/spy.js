export function MySpy() {
  this.calls = 0;
}

MySpy.prototype.fn = function () {
  return () => this.calls++;
}