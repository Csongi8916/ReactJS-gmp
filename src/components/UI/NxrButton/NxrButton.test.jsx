import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import NxrButton from '../../UI/NxrButton/NxrButton';
import { MySpy } from '../../../spy';

describe('NxrButton Snapshpt tests:', () => {
  test('Should render correctly NxrButton', () => {
    const tree = renderer.create(<NxrButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

/*describe('NxrButton UI tests:', () => {
  test('Should call click handler method', () => {
    const mySpy = new MySpy();
    const mockCallBack = mySpy.fn();
    const wrap = shallow(<NxrButton click={mockCallBack}>Search</NxrButton>);
    wrap.find('.NrxBtn').simulate('click');
    expect(mySpy.calls).toEqual(1);
  });
});*/