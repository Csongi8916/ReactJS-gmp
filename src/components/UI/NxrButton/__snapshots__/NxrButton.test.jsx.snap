// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NxrButton Snapshpt tests: Should render correctly NxrButton 1`] = `
<button
  className="NrxBtn"
  onClick={[Function]}
/>
`;
